File: fabric/src/main/java/nl/oxod/oxai/OxAIConfig.java
package nl.oxod.oxai;

import com.google.gson.Gson;
import com.google.gson.GsonBuilder;
import java.nio.file.Files;
import java.nio.file.Paths;
import java.io.File;

public class OxAIConfig {
  private static final java.nio.file.Path configPath = Paths.get("config", OxAI.MODID + ".json");
  private static final Gson gson = new GsonBuilder().setPrettyPrinting().create();

  private boolean enabled = true;
  private String trigger = "@ai";
  private String ollamaUrl = "http://localhost:11434";
  private String model = "llama3.2:3b";
  private String systemPrompt = "You are a helpful AI assistant in a Minecraft server. Keep responses concise and friendly.";
  private int maxTokens = 150;
  private double temperature = 0.7;
  private String responsePrefix = "[AI] ";
  private int maxMessageLength = 256;
  private boolean showErrors = true;
  private int requestTimeout = 60000;
  private int contextMessages = 5;

  public static OxAIConfig load() {
    File configFile = configPath.toFile();

    if (configFile.exists()) {
      try {
        String json = Files.readString(configPath);
        OxAIConfig config = gson.fromJson(json, OxAIConfig.class);
        OxAI.getLogger().info("Loaded config from " + configFile.getAbsolutePath());
        return config;
      } catch (Exception e) {
        OxAI.getLogger().error("Failed to load config, using defaults", e);
        OxAIConfig config = new OxAIConfig();
        config.save();
        return config;
      }
    } else {
      OxAI.getLogger().info("Config file not found, creating default");
      OxAIConfig config = new OxAIConfig();
      config.save();
      return config;
    }
  }

  public void save() {
    try {
      File configFile = configPath.toFile();
      configFile.getParentFile().mkdirs();

      String json = gson.toJson(this);
      Files.writeString(configPath, json);

      OxAI.getLogger().info("Saved config to " + configFile.getAbsolutePath());
    } catch (Exception e) {
      OxAI.getLogger().error("Failed to save config", e);
    }
  }

  public boolean isEnabled() {
    return enabled;
  }

  public void setEnabled(boolean enabled) {
    this.enabled = enabled;
  }

  public String getTrigger() {
    return trigger;
  }

  public void setTrigger(String trigger) {
    this.trigger = trigger;
  }

  public String getOllamaUrl() {
    return ollamaUrl;
  }

  public void setOllamaUrl(String ollamaUrl) {
    this.ollamaUrl = ollamaUrl;
  }

  public String getModel() {
    return model;
  }

  public void setModel(String model) {
    this.model = model;
  }

  public String getSystemPrompt() {
    return systemPrompt;
  }

  public void setSystemPrompt(String systemPrompt) {
    this.systemPrompt = systemPrompt;
  }

  public int getMaxTokens() {
    return maxTokens;
  }

  public void setMaxTokens(int maxTokens) {
    this.maxTokens = maxTokens;
  }

  public double getTemperature() {
    return temperature;
  }

  public void setTemperature(double temperature) {
    this.temperature = temperature;
  }

  public String getResponsePrefix() {
    return responsePrefix;
  }

  public void setResponsePrefix(String responsePrefix) {
    this.responsePrefix = responsePrefix;
  }

  public int getMaxMessageLength() {
    return maxMessageLength;
  }

  public void setMaxMessageLength(int maxMessageLength) {
    this.maxMessageLength = maxMessageLength;
  }

  public boolean isShowErrors() {
    return showErrors;
  }

  public void setShowErrors(boolean showErrors) {
    this.showErrors = showErrors;
  }

  public int getRequestTimeout() {
    return requestTimeout;
  }

  public void setRequestTimeout(int requestTimeout) {
    this.requestTimeout = requestTimeout;
  }

  public int getContextMessages() {
    return contextMessages;
  }

  public void setContextMessages(int contextMessages) {
    this.contextMessages = contextMessages;
  }
}

File: src/main/kotlin/nl/oxod/oxai/OxAIConfig.kt
package nl.oxod.oxai

import com.google.gson.GsonBuilder
import java.nio.file.Files
import java.nio.file.Paths

data class OxAIConfig(
  var enabled: Boolean = true,
  var trigger: String = "@ai",
  var ollamaUrl: String = "http://localhost:11434",
  var model: String = "llama3.2:3b",
  var systemPrompt: String = "You are a helpful AI assistant in a Minecraft server. Keep responses concise and friendly.",
  var maxTokens: Int = 150,
  var temperature: Double = 0.7,
  var responsePrefix: String = "[0xAI] ",
  var maxMessageLength: Int = 256,
  var showErrors: Boolean = true,
  var requestTimeout: Int = 30000,
  var contextMessages: Int = 5
) {
  companion object {
    private val configPath = Paths.get("config", "${OxAI.MODID}.json")
    private val gson = GsonBuilder().setPrettyPrinting().create()

    fun load(): OxAIConfig {
      val configFile = configPath.toFile()

      return if (configFile.exists()) {
        try {
          val json = Files.readString(configPath)
          gson.fromJson(json, OxAIConfig::class.java).also {
            OxAI.logger.info("Loaded config from ${configFile.absolutePath}")
          }
        } catch (e: Exception) {
          OxAI.logger.error("Failed to load config, using defaults", e)
          OxAIConfig().also { it.save() }
        }
      } else {
        OxAI.logger.info("Config file not found, creating default")
        OxAIConfig().also { it.save() }
      }
    }
  }

  fun save() {
    try {
      val configFile = configPath.toFile()
      configFile.parentFile.mkdirs()

      val json = gson.toJson(this)
      Files.writeString(configPath, json)

      OxAI.logger.info("Saved config to ${configFile.absolutePath}")
    } catch (e: Exception) {
      OxAI.logger.error("Failed to save config", e)
    }
  }
}
